packages/flutter/lib/src/widgets/actions.dart

```diff
@@ -613,36 +618,68 @@ class _FocusableActionDetectorState extends State<FocusableActionDetector> {
   void _handleMouseEnter(PointerEnterEvent event) {
     assert(widget.onShowHoverHighlight != null);
     if (!_hovering) {
-      // TODO(gspencergoog): remove scheduleMicrotask once MouseRegion event timing has changed.
-      scheduleMicrotask(() { setState(() { _hovering = true; _handleShowHoverHighlight(); }); });
+      _mayTriggerCallback(task: () {
+        setState(() { _hovering = true; });
+      });
     }
   }
 
   void _handleMouseExit(PointerExitEvent event) {
     assert(widget.onShowHoverHighlight != null);
     if (_hovering) {
-      // TODO(gspencergoog): remove scheduleMicrotask once MouseRegion event timing has changed.
-      scheduleMicrotask(() { setState(() { _hovering = false; _handleShowHoverHighlight(); }); });
+      _mayTriggerCallback(task: () {
+        setState(() { _hovering = false; });
+      });
     }
   }
 
   bool _focused = false;
   void _handleFocusChange(bool focused) {
     if (_focused != focused) {
-      setState(() {
-        _focused = focused;
-        _handleShowFocusHighlight();
-        widget.onFocusChange?.call(_focused);
+      _mayTriggerCallback(task: () {
+        setState(() {
```

Comment:
Now that I look at this, I'm not sure that it even needs to call `setState`, since it doesn't use the value of _focused in its `build` function.

Reply:
I was thinking about this too, although `setState` does mark its children as dirty, so you might want this?

Reply:
Also, if none of the properties are states, is there really a reason to make `FocusableActionDetector` a widget any more?

Reply:
No, I think that it definitely is a good convenience widget to combine several concepts for desktop support, so it still has value.